src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -69,88 +89,159 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
                                           0,
                                           nullptr));
 
+        THROW_LAST_ERROR_IF_NULL(_hOutputThread);
+
         _connected = true;
     }
 
+    // Method description:
+    // - ascribes to the ITerminalConnection interface
+    // - handles the different possible inputs in the different states
+    // Arguments:
+    // the user's input
     void AzureConnection::WriteInput(hstring const& data)
     {
         if (!_connected || _closing.load())
         {
             return;
         }
 
-        // This case only happens when the user has multiple tenants
-        // and we require their input for which tenant they want
-        if (_needInput)
+        // Parse the input differently depending on which state we're in
+        switch (_state)
         {
-            int tenantNum = -1;
-            while (tenantNum < 0 || tenantNum >= _maxSize)
+        // The user has stored connection settings, let them choose one of them, create a new one or remove all stored ones
+        case State::accessStored:
+        {
+            std::string s = winrt::to_string(data);
+            int storeNum = -1;
+            try
             {
-                if (_closing.load())
+                storeNum = std::stoi(s);
+            }
+            catch (...)
+            {
+                if (s != "n" && s != "r")

```

Comment:
This sort of test will only test the first character in the string, ignoring everything after it. Is that what you want? Or do you also want to guarantee that it's a 1-length string?

Reply:
Its not possible for the string to be longer than 1-length, the function is called the moment the user types anything (it does not wait for the user to press 'enter' or something)